### PR TITLE
Iris Docs: Dark theme ready

### PR DESCRIPTION
**Pull request by tkknight**
### Description
Updates the Iris docs to be dark theme ready.  This is a feature that the [pydata-sphinx-theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html) supports.

PR build of the docs: https://scitools-iris--5299.org.readthedocs.build/en/5299/index.html

#### Changes include

- Enabled the dark theme, this now adds an icons that toggles auto/dark/light (top right corner ish)
- ``svg`` images have an increased border to so when there is a dark background it looks better
- All images are transparent where appropriate
- datatables (used for the issue voting table)
   - Updated the datatables version (no visible changes here but I may have made the table dark mode friendly as it was not before hand)
   - Added some useful defaults for the table presentation
- Updated the dropdown syntax when using the code icon
- Specific a `dark-light` class where the figure/image is transparent, otherwise the theme will add a white background to the transparent image when using dark more.  Sometimes this is ok though so the class has not been used everywhere.

#### Minor issue outstanding

The Iris [Voted Issues](https://scitools-iris--5299.org.readthedocs.build/en/5299/voted_issues.html#voted-issues-top) table looks ok apart from the pulldown and text entry that uses black text on the almost black background, making it very hard to read.  This is not a major issue and unless there are objections should not stop this PR.

I expect this issue will resolve itself when `pydata-sphinx-theme` is updated (and we unpin it) or when the `datatables` css is updated.

image

image

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by tkknight** [comment]
Ready for review.

File: docs/src/voted_issues.rst
Comment (trexfeathers) [inline docs/src/voted_issues.rst:1]: The poor rendering of the table is pretty frustrating, but I don't want to get in the way of the PR so I'll try to be less squeamish.

File: docs/src/_static/theme_override.css
Comment (trexfeathers) [inline docs/src/_static/theme_override.css:1]: Could you provide a reference / explanation for this change? I'm not sure where to look.

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks @tkknight. At least one fix to an image is needed, plus a few other comments.

File: docs/src/userguide/merge_and_concat.svg
[the remark this answers is not in the document]
Comment (tkknight) [reply docs/src/userguide/merge_and_concat.svg:1]: Fixed.

File: docs/src/voted_issues.rst
Comment (tkknight) [reply docs/src/voted_issues.rst:1]: I plan to check for datatables and theme updates that may fix this in future.

File: docs/src/userguide/cube_diagram.png
[the remark this answers is not in the document]
Comment (tkknight) [reply docs/src/userguide/cube_diagram.png:1]: deleted.

File: docs/src/_static/theme_override.css
Comment (tkknight) [reply docs/src/_static/theme_override.css:1]: If you view the live docs https://scitools-iris.readthedocs.io/en/latest/ and press F12, in the top right you will see a red cross with the number or html/css/script errors, mine shows 5.  The built docs https://scitools-iris.readthedocs.io/en/latest/ for this PR corrects a path so it is found correctly, now showing only 4 errors.  The remaining errors I suspect are due to the Met Office firewall.

**Review by trexfeathers — APPROVED** [review]

**Comment by trexfeathers** [comment]
Great, thanks @tkknight! Let's save some monitor energy and some retinas 💻👁 ...

**Pull request merged**